webpack/assets/javascripts/react_app/components/ForemanModal/index.js

```diff
@@ -1,3 +1,6 @@
 import ForemanModal from './ForemanModal';
+import reducer from './ForemanModalReducer';
+
```

Comment:
how about creating all the hooks here, and pass it to the component as props? so we can keep our separation convention

Reply:
@amirfefer Hooks must be used only inside a function component or in a custom Hook.  (But see the other comments for more discussion :yum: )

Reply:
How about this approach:

```js
// component/index.js

import { useSelector, useDispatch } from 'redux-react'
import Component from './Component'
import { selectAllData, selectSomeOtherData } from './ComponentSelectors'
import { hide } from './ComponentsActions'

const dispatch = useDispatch();

const ConnectedComponent = () =>
<Component
  allRecords=useSelector(selectAllData)
  otherData=useSelector(selectSomeOtherData)
  hide=dispatch(hide);
/>
```
  
With this we still gain a great separation between the markup and the logic, we use hooks and keep our conventions. It probably causes extra code writing, but such a separation I  believe makes the code cleaner and readable.

Reply:
```js
const dispatch = useDispatch();
```
This is a use of a hook outside a function component, so unfortunately it won't work